hazelcast/src/main/java/com/hazelcast/instance/impl/ClusterTopologyIntentTracker.java

```diff
@@ -0,0 +1,138 @@
+ *     <br/>- When a kubernetes pod is scheduled, "current replicas" immediately increases by 1 even though Hazelcast is just
+ *     started and not joined to any cluster.</li>
+ *     <li><b>Ready replicas</b>: replicas which are observed as "ready" by the managed runtime. "Ready" implies that the
+ *     managed runtime has queried the configured readiness probe and found the Hazelcast member to be ready (i.e. up & running
+ *     and ready to accept and serve requests). Notice that since "ready" state is based on a periodic check by the managed
+ *     runtime, its updates can lag for some time, depending on configuration (e.g. what is the period of readiness probe checks
+ *     etc). Example:
+ *     <br/>Kubernetes deletes a pod of a running cluster with specified replicas 3. Immediately
+ *     "current replicas" drops to 2, however "ready replicas" is still 3 until the readiness probe monitoring period passes
+ *     and the readiness check figures out the Hazelcast member is no longer ready.
+ *     <br/>
+ *     Also related to "readiness" definition: {@code NodeExtension#isReady}.</li>
+ * </ul>
+ */
+public interface ClusterTopologyIntentTracker {
+
+    int UNKNOWN = -1;
+
+    /**
+     * Process an update of the cluster topology. Each update carries information about
+     * the previous & current specified replicas count, the current number of replicas and ready replicas in the cluster.
+     * <br/>
+     * <b>Examples</b> (numbers in parentheses indicate (previous specified replica count, current specified replica count,
+     * current replicas, current ready replicas))
+     * <p>
+     *     A cluster with specified replica count 3 is starting up. This is expected to result in a series
+     *     of updates similar to the following:
+     *     <pre>{@code
+     *     (-1, 3, 0, 0)
+     *     (3, 3, 1, 0)
+     *     (3, 3, 1, 1)
+     *     (3, 3, 2, 1)
+     *     (3, 3, 2, 2)
+     *     (3, 3, 3, 2)
+     *     (3, 3, 3, 3)
+     *     }</pre>
+     * </p>
+     * <p>
+     *     Assuming user requests scaling up a running cluster of 3 members to 5, the following
+     *     updates are expected:
+     *     <pre>{@code
+     *     (3, 3, 3, 3)
+     *     (3, 5, 4, 3)
+     *     (5, 5, 4, 4)
+     *     (5, 5, 5, 4)
+     *     (5, 5, 5, 5)
+     *     }</pre>
+     * </p>
+     * Notice that actual updates may differ (eg duplicate notifications of intermediate states may be received).
+     *
+     * @param previousSpecifiedReplicaCount   previous specified replicas count
+     * @param currentSpecifiedReplicaCount    current specified replicas count
+     * @param readyReplicasCount              number of ready replicas
+     * @param currentReplicasCount            number of current replicas
+     *
+     * @see NodeExtension#isReady()
+     */
+    void update(int previousSpecifiedReplicaCount, int currentSpecifiedReplicaCount, int readyReplicasCount,
+                int currentReplicasCount);
```

Comment:
Again, a restart scenario would be much useful to understand event coalesced events.